### PR TITLE
Add unittest for DomainParticipantFactory get_instance method

### PR DESCRIPTION
**Pull request by laura-eprosima**

### CODE REVIEW

File: test/unittest/dds/participant/ParticipantTests.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+TEST(ParticipantTest, DomainParticipantFactoryGetInstance)
+{
+    DomainParticipantFactory* factory = DomainParticipantFactory::get_instance();
+
+    ASSERT_NE(factory, nullptr);
```
Comment (MiguelCompany) [inline test/unittest/dds/participant/ParticipantTests.cpp]: I would also check that a second call gives the same result. This is a singleton, right?

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1602/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1368/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1598/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1605/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1371/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1601/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1827/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1574/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1821/)

**Pull request merged**